geopandas/geodataframe.py

```diff
@@ -445,14 +444,36 @@ def crs(self):
     @crs.setter
     def crs(self, value):
         """Sets the value of the crs"""
-        if self._geometry_column_name not in self:
+        if self._geometry_column_name is None:
+            raise ValueError(
+                "Assigning CRS to a GeoDataFrame without a geometry column is not "
+                "supported. Use GeoDataFrame.set_geometry to set the active "
+                "geometry column.",
+            )
+
+        if hasattr(self.geometry.values, "crs"):
+            self.geometry.values.crs = value
+        else:
+            # column called 'geometry' without geometry
+            raise ValueError(
+                "Assigning CRS to a GeoDataFrame without an active geometry "
+                "column is not supported. Use GeoDataFrame.set_geometry to set "
+                "the active geometry column.",
+            )
+
+        if (
+            self._geometry_column_name is None
+            and self._geometry_column_name not in self
```

Comment:
Is this something that will happen in practice? (do we have a test for that?)

Reply:
Actually, it this check still needed? Didn't we already raise an error above in case of `if self._geometry_column_name is None:` ?